### PR TITLE
DB WEEK 2 HW added

### PR DESCRIPTION
**Pull request by Vahablotfi**

### CODE REVIEW

File: databases/week2/db week 2/DB week2 HW [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+ ORDER BY COUNT(created);
+
+ -- Part 4: Creating a database
+
+ CREATE DATABASE Fitness_DB;
+-- For this part , I used a sport club which provides different 
+-- classes and different trainers.
+
+
+CREATE TABLE `fitness_class` (
```
Comment (Vahablotfi) [inline databases/week2/db week 2/DB week2 HW:176]: Screenshot 2022-02-17 at 14 56 41

File: databases/week2/db week 2/DB week2 HW [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+-- Change a task due date
+	UPDATE task 
+    SET due_date = '2025-8-20'
+    WHERE id = 35;
+
+-- Change a task status
+
+	UPDATE task 
+    SET status_id = 3
+    WHERE status_id != 3
```
Comment (martamatos) [inline databases/week2/db week 2/DB week2 HW:23]: You're supposed to change the status of only one task, not all tasks that have the status_id=3 ;p

File: databases/week2/db week 2/DB week2 HW [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+    LIMIT 1;
+-- Mark a task as complete
+	
+    	UPDATE task 
+    SET status_id = 
+    (SELECT id
+FROM status
+WHERE status.name='Done')
+    WHERE status_id !=
```
Comment (martamatos) [inline databases/week2/db week 2/DB week2 HW:32]: Same comment as above ;p

File: databases/week2/db week 2/DB week2 HW [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+ WHERE id = 47 ;
+  
+  Select *
+    FROM task;
+-- I added some task, but  I had to delet them , and after that 
+--  whatever task I made got an id which was right after the 
+--deleted tasks id (you can see the id after 35 is 46 , 
+--instead of 36,because I deleted ten task), 
+--I couldn't find a way to fix that ,
+-- I would appreciate a guide for that issue.
```
Comment (martamatos) [inline databases/week2/db week 2/DB week2 HW:51]: It's just the way it works. 
However, you can define the id when you insert a task, and in that case you can set it to 36.
I'd say it's not worth it in general, just let MySQL auto increment it xD

File: databases/week2/db week 2/DB week2 HW [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+     
+ 
+-- Student: with the columns: id, name, email, phone, class_id (foreign key)
+ CREATE TABLE `Student`(
+ `id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+ `name` VARCHAR(50) NOT NULL,
+ `email` VARCHAR(50),
+ `phone` int(15) NOT NULL,
+ `class_id` int (10) unsigned NOT NULL ,
+  CONSTRAINT `fk_class` FOREIGN KEY (`class_id`) REFERENCES `Class` (`id`) ON DELETE  CASCADE ON UPDATE CASCADE
```
Comment (martamatos) [inline databases/week2/db week 2/DB week2 HW:86]: As in the homework for week3, if you delete a class with id 2, all students with a class_id=2 will also be deleted ;)

File: databases/week2/db week 2/DB week2 HW [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+FROM user u
+JOIN user_task  ut ON u.id=ut.user_id
+JOIN task t ON t.id = ut.task_id
+ WHERE u.email LIKE '%@spotify.com';
+
+
+
+-- Get all the tasks for 'Donald Duck' with status 'Not started'
+SELECT 
+t.title AS task_title,
```
Comment (martamatos) [inline databases/week2/db week 2/DB week2 HW:127]: I know `t` is shorter than `task`, but it's also harder to read, so when choosing variable/function/etc names, keep in mind that the code needs to be easier to read. In real life, someone will have to maintain it, and that someone might even be future you ;p

**Review by martamatos — APPROVED** [review]
Good work Vahab! I left some small comments but you don't need to change anything, just merge it :)

**Pull request merged**